app/src/main/java/com/a0xffffffff/dinesum/FirebaseManager.java

```diff
@@ -197,11 +207,12 @@ public void onCancelled(DatabaseError databaseError) { //update UI here if error
             public void onDataChange(DataSnapshot dataSnapshot) { //something changed!
                 ArrayList<Request> userReservations = new ArrayList<Request>();
                 for (DataSnapshot requestSnapshot : dataSnapshot.getChildren()) {
-                    Request newRequest = FirebaseManager.parseJson(requestSnapshot);
+                    Request newRequest = parseJson(requestSnapshot);
                     Log.d("Init user reservations", "requestID: " + newRequest.getRequestID());
                     userReservations.add(newRequest);
                 }
                 // save nearby requests
+                Collections.reverse(userReservations);
                 RequestTracker.getInstance().setUserReservations(userReservations);
             }
```

Comment:
Similarly, do we have to call onReserverRequestsReady() here?

Reply:
same here